[staging] Build failure: pkgsMusl.busybox

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

nix build nixpkgs/staging#pkgsMusl.busybox

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
[...]
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j8 SHELL=/nix/store/519jm22hm0harapmwwxgvgjwaav0hdjq-bash-5.3p9/bin/bash SKIP_STRIP=y
make: *** [Makefile:858: include/config/MARKER] Illegal instruction (core dumped)
```

### Additional context

good (master): [85b9eb70a313f3a908b036c6c06a1e194df72ca4] tsgolint: 0.18.0 -> 0.18.1
bad (staging): [b80b3ebc19cda67252c2ec54a87e75a1b797d227] gdb: fix build for musl


### System metadata


 - system: `"x86_64-linux"`
 - host os: `Linux 6.19.10, NixOS, 26.05 (Yarara), 26.05.20260327.f5f805f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.96.0-dev-pre20260327-dev_001e3fd
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/yuka/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/yuka/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/yuka/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/yuka/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/npknmg04r430rgz2mnxi2nppgcqz76ia-lix-2.96.0-dev-pre20260327-dev_001e3fd/share`
 - nixpkgs: `/nix/store/wy3ykschab4gv94skf5i27ya08l84qds-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/busybox/build-system-buffer-overflow.patch
+++ b/pkgs/os-specific/linux/busybox/build-system-buffer-overflow.patch
@@ -1,0 +1,27 @@
+From 3cf1ca7491bd5b6680e80355d76442ae14db681e Mon Sep 17 00:00:00 2001
+From: Alyssa Ross <hi@alyssa.is>
+Date: Sun, 29 Mar 2026 13:18:09 +0200
+Subject: [PATCH] build system: fix potential buffer overflow
+
+This could potentially write one byte past the end of line.
+Identified by fortify-headers.
+---
+ scripts/basic/split-include.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/scripts/basic/split-include.c b/scripts/basic/split-include.c
+index 6ef29195e..93011d511 100644
+--- a/scripts/basic/split-include.c
++++ b/scripts/basic/split-include.c
+@@ -195,7 +195,7 @@ int main(int argc, const char * argv [])
+ 	ERROR_EXIT( "find" );
+ 
+     line[0] = '\n';
+-    while (fgets(line+1, buffer_size, fp_find))
++    while (fgets(line+1, buffer_size-1, fp_find))
+     {
+ 	if (strstr(list_target, line) == NULL)
+ 	{
+-- 
+2.53.0
+

--- a/pkgs/os-specific/linux/busybox/default.nix
+++ b/pkgs/os-specific/linux/busybox/default.nix
@@ -90,6 +90,8 @@ stdenv.mkDerivation rec {
       excludes = [ "networking/httpd_ratelimit_cgi.c" ]; # New since release.
       hash = "sha256-Msm9sDZrVx7ofunnvnTS73SPKUUpR3Tv5xZ/wBd+rts=";
     })
+    # https://lists.busybox.net/pipermail/busybox/2026-March/092010.html
+    ./build-system-buffer-overflow.patch
   ]
   ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform) ./clang-cross.patch;
 

